web/react-web/src/Graph/Graph.js

```diff
@@ -0,0 +1,241 @@
+
+                let x = null;
+                if (xAxisRelative) {
+                    x = index + 1;
+                } else {
+                    x = Date.parse(date);
+                }
+
+                let marker = null;
+                let enabled = null;
+
+                if (x === maxXValue) {
+                    enabled = true;
+                    marker = {
+                        symbol: 'triangle-down',
+                            enabled: true,
+                            radius: 8
+                    };
+                }
+
+                if (x === predictionXValue) {
+                    enabled = true;
+                    marker = {
+                        symbol: 'diamond',
+                        enabled: true,
+                        radius: 8
+                    };
+                }
+
+                return {
+                    enabled, x, y, marker
+                };
+            });
+
+            series.push({
+                type: 'line',
+                dashStyle,
+                name,
+                marker: {
+                    enabled: data.type !== 'prediction'
+                },
+                data: lineData,
+                color: colors[series.length % colors.length],
+                zones,
+                zoneAxis: 'x'
+            });
+        });
+
+        finalOptions.title = {
+            text: `Active cases and prediction for ${countries.join(', ')}`
+        };
+
+        finalOptions.series = series;
+
+        switch (axesType) {
+            default:
+            case AXES_LINEAR:
+                finalOptions.xAxis.type = 'linear';
+                finalOptions.yAxis.type = 'linear';
+                break;
+            case AXES_LOG:
+                finalOptions.yAxis.type = 'logarithmic';
+                finalOptions.xAxis.type = 'linear';
+                break;
+            case AXES_LOG_LOG:
+                finalOptions.xAxis.type = 'logarithmic';
+                finalOptions.yAxis.type = 'logarithmic';
+                break;
+        }
+
+        if (xAxisRelative) {
+            finalOptions.xAxis.title.text = 'Day since relevant number of cases';
+
+            finalOptions.tooltip.formatter = function() {
+                //https://api.highcharts.com/highcharts/tooltip.formatter
+                if (this.points) {
+                    //shared tooltip, we need extract data from every serie
+                    return this.points.reduce((s, point) => {
+                        const name = point.series.name;
+                        return `${s}<br/><strong>${name}:</strong> ${Math.round(point.y)}`;
+                    }, `<div style="text-align: center"><strong>Day ${this.x}</strong></div>`);
+                } else {
+                    const name = this.point.series.name;
+                    return `<div style="text-align: center"><strong>Day ${this.x}</strong></div><br/><strong>${name}:</strong> ${Math.round(this.point.y)}`;
```

Comment:
An idea what might make this code less repetitive: Could we define a function
```
xaxis_text(x, xAxisRelative)
```
That returns `Day ${this.x}` or `${date.toLocaleDateString()}`?

Reply:
OMG, right! :)